### ISSUE TITLE
best solution to automatically detect (and rotate) barcode (QRCode, DataMatrix) on normal image

### ISSUE THREAD

**Issue by DanielRuf** [zxing-js/browser#49]
ZXing has issues to detect only the barcode on an image if this is not perfectly cropped and rotated.

Does anyone know a good solution that works also for QCode and DataMatrix?

**Comment by odahcam** [comment]
Chrome's image recognition API may help.

**Comment by DanielRuf** [comment]
> Chrome's image recognition API may help.

If you mean the barcode reader API, this is not available in webkit / all platforms.

**Comment by odahcam** [comment]
Have you tried the try-harder hint with this library?

**Comment by DanielRuf** [comment]
> try-harder hint

I saw this but it is not really explained what it does. It seems to rotate the image a bit. But this does no cropping, skewing and so on, correct?

https://github.com/zxing-js/library/issues/303

https://github.com/zxing-js/browser/commit/ec12fd9aeb4a4e0baf444ad1684971929974de1b

In our case users upload photos of documents which contain the actual QR codes. So it is not perfectly cropped, rotated, skewed or aligned.

We already tried it with some tests and so far images only work if they are perfectly cut / cropped.
